plugin/build.gradle

```diff
@@ -75,10 +78,14 @@ sourceSets {
     }
 }
 
-compileKotlin {
-    kotlinOptions {
-        freeCompilerArgs += "-Xopt-in=kotlin.RequiresOptIn"
-        apiVersion = "1.4" // the version of Kotlin embedded in Gradle
+tasks.named("compileKotlin", KotlinCompilationTask.class) {
+    compilerOptions {
+        optIn.addAll(
+                "kotlinx.benchmark.gradle.internal.KotlinxBenchmarkPluginInternalApi",
+                "kotlin.RequiresOptIn",
+        )
+        //noinspection GrDeprecatedAPIUsage
+        apiVersion = KotlinVersion.KOTLIN_1_4 // the version of Kotlin embedded in Gradle
```

Comment:
btw, makes sense also to limit `languageVersion` to `KOTLIN_1_4` or `KOTLIN_1_5`. Currently it's being compiled with the default LV equal to the compiler version, and it even works with old Gradle versions, but is not guaranteed to work. It does not directly relate to your changes, so you can do it separately.